### DISCUSSION TITLE
Satori

### DISCUSSION THREAD

**Discussion by codingluke** [marp-team/marp#362] [Ideas] [answered]
I just found https://github.com/vercel/satori and thought It could be very interessting for marp. Of course, it might be a very big refactoring. I just wantet to let you know :)

**Comment by yhatt** [comment, marked as answer]
Satori is only supports minimum subset of HTML and CSS, and seems not to aim to make full-browser rendering features. Thus, we don't think Satori can adopt as an alternative slide renderer, for the present.

On the other hand, the slide generated from Marpit framework may be able to convert through Satori, by passing a lot of transformations. For instance:

1. Generate a plain slide using Marpit framework (Marp Core cannot use due to incompatible inline SVG mode for now)
1. Convert CSS to inline styles that were assigned to HTML elements (e.g. [`ultrahtml@1.0.0`](https://github.com/natemoo-re/ultrahtml/releases) w/ `inline` transformer)
1. Pass generated HTML in 2 to [`satori-html`](https://github.com/natemoo-re/satori-html) and create Satori compatible object.
1. Generate SVG from an object created in 3 by using Satori.

> **Warning** Slide CSS generated by Marpit may require postprocessing to remove uncompatible pseudo-elements with ultrahtml and Satori.

This screenshot is my evaluation about Marpit slide rendering through Satori. A very simple Markdown + limited CSS layout can reproduce as a SVG image rendered by Satori. :point_down: 

![localhost_5173_ (1)](https://user-images.githubusercontent.com/3993388/199074174-e9cdca31-650d-49aa-892e-0776181df477.png)


Developer who interested in this idea would be able to try and evaluate by following above transformation steps.
